### PR TITLE
Bugfix in ATSAME5x network interface. Incorrect detection of ICMP

### PR DESCRIPTION
**Pull request by cruxic**
A fix for issue #1193. 

Note: although I have an ATSAME53 board, I have not compiled or executed the code in this pull-request. My own project uses a customized version of this code.

I also noticed that this same bug exists in the MSP432 driver. This pull-request does not fix that.

### CODE REVIEW

**Review by jasonpcarroll — COMMENTED** [review]
Looks fine to me. I do not _currently_ have an ATSAME5x with me - would you mind running this version of your code in your project and verifying that it works?

**Review by htibosch — COMMENTED** [review]
Please have a look at this slightly modified version:

[NetworkInterface.new.zip](https://github.com/user-attachments/files/17173668/NetworkInterface.new.zip)

File: source/portable/NetworkInterface/ATSAME5x/NetworkInterface.c [hunk 1]
```diff
@@ -335,16 +348,12 @@ static void prvEMACDeferredInterruptHandlerTask( void * pvParameters )
                     #if ( ipconfigDRIVER_INCLUDED_RX_IP_CHECKSUM == 1 )
                     {
                         /* the Atmel SAM GMAC peripheral does not support hardware CRC offloading for ICMP packets.
-                         * It must therefore be implemented in software. */
-                        pxIPPacket = ( IPPacket_t const * ) pxBufferDescriptor->pucEthernetBuffer;
-
-                        if( pxIPPacket->xIPHeader.ucProtocol == ( uint8_t ) ipPROTOCOL_ICMP )
-                        {
+                        * It must therefore be implemented in software. */
+                        if ( isICMP(pxBufferDescriptor) ) {
```
Comment (htibosch) [inline source/portable/NetworkInterface/ATSAME5x/NetworkInterface.c:352]: The `if` statement expects a boolean value. The function `isICMP()` returns an integer value. The only way to satisfy the compiler is as follows:
~~~
if( isICMP( pxBufferDescriptor ) == pdTRUE )
{
}
~~~

**Comment by cruxic** [comment]
> Hi @cruxic , Thank you for this PR. Looks fine to me. I do not _currently_ have an ATSAME5x with me - would you mind running this version of your code in your project and verifying that it works?

I'm not able to execute this code directly, because I'm using a heavily modified version of the code. The code included in FreeRTOS+TCP is based on ATMEL ASF libraries, which is deprecated since Microchip acquired ATMEL.

**Review by tony-josi-aws — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**